app/controllers/authentications_controller.rb

```diff
@@ -33,7 +33,7 @@ def destroy
     if @authentication
       if current_user.authentications.count == 1 and current_user.encrypted_password.blank?
         # Bryan: TESTED
-        flash[:alert] = 'Bad idea!'
+        flash[:alert] = 'Failed to unlink GitHub. Please use another provider for login or reset password.'
```

Comment:
this is great - I might chance to "Failed to unlink GitHub. To unlink GitHub please first connect your account to another provider (e.g. Google+) or reset your password to create email authentication.", but even with your text above this is a great improvement ... I mean we should double check that resetting password is an option for the user if they only ever signed up via a 3rd party provider.

The key thing to thing about here @touchps is that while we internally might understand what a "provider" is, there's no reason that our users will be thinking in terms of that word. That's the problem with so many error messages - they are constructed from the developers point of view and using the developers vocabulary; can be very confusing for user ...